python-threatexchange/threatexchange/exchanges/signal_exchange_api.py

```diff
@@ -169,19 +169,46 @@ def naive_fetch_merge(
                 old[k] = new_v
 
     @classmethod
-    @abstractmethod
     def naive_convert_to_signal_type(
         cls,
         signal_types: t.Sequence[t.Type[SignalType]],
         collab: TCollabConfig,
         fetched: t.Mapping[state.TUpdateRecordKey, state.TUpdateRecordValue],
-    ) -> t.Dict[t.Type[SignalType], t.Dict[str, state.TFetchedSignalMetadata]]:
+    ) -> t.Mapping[t.Type[SignalType], t.Dict[str, state.TFetchedSignalMetadata]]:
```

Comment:
It was dict before because this isn't supposed to store internal state, which means that ownership of all objects can be transferred to the caller(Dict is mutable, Map is immutable).